### PR TITLE
feat: remove notifications once the post has been approved/rejected/deleted

### PR DESCRIPTION
**Pull request by sshanzel**
Reference: https://dailydotdev.slack.com/archives/C07GFU7DX8X/p1732006742496929

Changes:
- Renamed some files to follow current convention.
- Realized we are missing some tests about what would the actual notification generated look like.
  - Found out we were missing the source avatars.
- Clear the notifications sent when the post has been approved/rejected/deleted.

### CODE REVIEW

File: __tests__/notifications/index.ts [hunk 1]
```diff
@@ -1253,6 +1258,158 @@ describe('storeNotificationBundle', () => {
     ]);
   });
 
+  it('should generate source_post_submitted notification', async () => {
```
Comment (sshanzel) [inline __tests__/notifications/index.ts:1261]: Realized we were missing these 3 tests, hence, added here now along with the fix.

File: src/notifications/common.ts [hunk 2]
```diff
@@ -283,11 +285,17 @@ export const generateUserNotificationUniqueKey = ({
...
-  [SourcePostModerationStatus.Pending]: NotificationType.SourcePostSubmitted,
+export const cleanupSourcePostModerationNotifications = async (
+  con: ConnectionManager,
+  post: ChangeObject<SourcePostModeration>,
+) => {
+  if (!post?.id) {
+    return;
+  }
+
+  await con.getRepository(NotificationV2).delete({
```
Comment (capJavert) [inline src/notifications/common.ts:296]: I am not sure we want to delete notification, they are usually immutable and can reference other stuff, maybe introduce soft delete cc: @idoshamun ?

Comment (idoshamun) [reply src/notifications/common.ts:296]: let me search for some references. I know for sure that when we delete a post we also hide the notifications. Not sure if we delete or soft delete

Comment (idoshamun) [reply src/notifications/common.ts:296]: So I guess we do delete notifications 😂

Comment (capJavert) [reply src/notifications/common.ts:296]: :triggered:

**Review by capJavert — APPROVED** [review]

**Pull request merged**